packages/app/src/cli/services/transform.ts

```diff
@@ -0,0 +1,258 @@
+import * as jscodeshift from 'jscodeshift/src/Runner.js'
+import {cwd, dirname, joinPath, resolvePath} from '@shopify/cli-kit/node/path'
+import {isClean} from '@shopify/cli-kit/node/git'
+import {glob} from '@shopify/cli-kit/node/fs'
+import {IPluginInfo, PluginManager} from 'live-plugin-manager'
+import {
+  renderAutocompletePrompt,
+  renderTextPrompt,
+  renderWarning,
+  renderSuccess,
+  renderTasks,
+} from '@shopify/cli-kit/node/ui'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import fs from 'fs'
+import {fileURLToPath} from 'url'
+
+const packages = ['@shopify/polaris-codemods']
+
+interface TransformOptions {
+  /** Files or directory to transform */
+  include?: string
+
+  /** The package to run transforms for, \@scope/package. */
+  package?: string
+
+  /** The name of transform */
+  transform?: string
+
+  /** If true, no code will be edited */
+  dry?: boolean
+
+  /** Show more information about the transform process */
+  verbose?: boolean
+
+  /** If true, the changed output is printed for comparison */
+  print?: boolean
+
+  /** If true, bypass Git safety checks and forcibly run transform */
+  force?: boolean
+}
+
+export async function transform(options: TransformOptions) {
+  if (!((await isClean(cwd())) || options.dry)) {
+    if (options.force) {
+      renderWarning({
+        headline: 'Forcibly continuing',
+        body: 'You have opted to bypass the Git safety check. This may result in lost work.',
+      })
+    } else {
+      renderWarning({
```

Comment:
Should this be a warning or an error? Feels like we didn't do the thing they asked for, so I'd expect an `AbortError`.